app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
+  end
+
+
+  def has_company?
+    has_membership_application? && (membership_applications.select { |app| app.company && app.company.persisted? }).count > 0
+  end
+
+
+  def membership_application
+    has_membership_application? ? membership_applications.last : nil
+  end
+
+
+  def company
+    has_company? ? membership_application.company : nil
+  end
```

Comment:
I think it would be good to firm up the model relationships very soon.  Otherwise, we might get tangled up in knots with associations that don't make sense in the real world.